src/components/Select/Select.tsx

```diff
@@ -1,574 +1,234 @@
+            {children}
+          </ComboboxProvider>
+        </SelectPopoverRoot>
+        {hasError && <Error>{error}</Error>}
+      </FormElementContainer>
       {label && (
         <Label
           htmlFor={id ?? defaultId}
           disabled={disabled}
-          error={!!error && typeof error !== "undefined"}
+          error={hasError}
         >
           {label}
         </Label>
       )}
     </FormRoot>
   );
 };
-interface GroupProps extends Omit<DivProps, "value" | "heading"> {
-  heading?: ReactNode;
-  value?: string;
-}
 
-const SelectGroup = styled(Command.Group)`
-  display: flex;
-  flex-direction: column;
-  align-items: flex-start;
-  justify-content: center;
-  width: var(--radix-popover-trigger-width);
-  padding: 0;
-  gap: 0.5rem;
-  &[aria-selected] {
-    outline: none;
-  }
+interface TriggerProps extends Omit<HTMLAttributes<HTMLButtonElement>, "id"> {
+  placeholder?: string;
+}
 
-  ${({ theme }) => `
-    font: ${theme.click.genericMenu.item.typography.label.default};
-    background: ${theme.click.genericMenu.item.color.background.default};
-    color: ${theme.click.genericMenu.item.color.text.default};
-    &[data-highlighted] {
-      font: ${theme.click.genericMenu.item.typography.label.hover};
-      background: ${theme.click.genericMenu.item.color.background.hover};
-      color:${theme.click.genericMenu.item.color.text.hover};
-    }
-    &[data-state="checked"] {
-      background:${theme.click.genericMenu.item.color.background.active};
-      color:${theme.click.genericMenu.item.color.text.active};
-      font: ${theme.click.genericMenu.item.typography.label.active};
-    }
-    &[data-disabled] {
-      background:${theme.click.genericMenu.item.color.background.disabled};
-      color:${theme.click.genericMenu.item.color.text.disabled};
-      font: ${theme.click.genericMenu.item.typography.label.disabled};
-      pointer-events: none;
-    }
-  `};
-  [cmdk-group-heading] {
-    display: flex;
-    width: 100%;
-    flex-direction: column;
-    max-width: calc(var(--radix-popover-trigger-width) - 24px);
-    overflow: hidden;
-    white-space: nowrap;
-    text-overflow: ellipsis;
-    ${({ theme }) => `
-     font: ${theme.click.genericMenu.item.typography.sectionHeader.default};
-     color: ${theme.click.genericMenu.item.color.text.muted};
-     padding: ${theme.click.genericMenu.item.space.y} ${theme.click.genericMenu.item.space.x};
-     gap: ${theme.click.genericMenu.item.space.gap};
-     border-bottom: 1px solid ${theme.click.genericMenu.item.color.stroke.default};
-   `}
-  }
-  [cmdk-group-items] {
-    width: 100%;
-  }
-  &[hidden] {
-    display: none;
-    [cmdk-group-heading] {
-      display: none;
-    }
-  }
-`;
+const Trigger = forwardRef<HTMLButtonElement, TriggerProps>(
+  ({ placeholder = "Select an option", onClick: onClickProp, ...props }, ref) => {
+    const [, forceUpate] = useState(0);
+    const { disabled, id, hasError, selectedValues, updateSearch, getValueProps } =
```

Comment:
@serdec @fneves @gjones Do you think elements such as `disabled, id, hasError` should be moved to a different context and also I should create a separate component with the useCombobox for the value only?